sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSessionCatalogSuite.scala

```diff
@@ -71,4 +71,12 @@ class DataSourceV2SQLSessionCatalogSuite
       sql(s"CREATE EXTERNAL TABLE t (i INT) USING $v2Format TBLPROPERTIES($prop)")
     }
   }
+
+  test("SPARK-49152: partition columns should be put at the end") {
```

Comment:
The test case name is a little strange. Is this a correct reproducer for `DelegatingCatalogExtension` issue, @amaliujia ?

Reply:
This is the Spark behavior with the built-in catalog. A `DelegatingCatalogExtension` shouldn't change it.

Strictly speaking, this shouldn't be the only issue, as there might be other subtle differences between v1 and v2 CREATE TABLE command, or other commands.

Reply:
Yeah as @cloud-fan have mentioned, this verifies a specific case so we make sure the behavior is not changed. And there might be a list to verify though.

Reply:
Got it. It makes sense. Thank you for the explanation.